review/get_review_ids.py

```diff
@@ -0,0 +1,212 @@
+        # record.module_state_key takes the form:
+        # block-v1:{course_id}+type@problem+block@{problem_id}
+        problem_id = str(record.module_state_key).split("@")[-1]
+
+        # Actual logic regarding the record should go here
+        state = json.loads(record.state)
+
+        # The key 'selected' shows up if a problem comes from a
+        # library content module. These cause issues so we skip this.
+        # Issue: Library content contains problems but the CSM brings up
+        # the library content and not the problems within
+        if 'selected' in state:
+            continue
+
+        # parent takes the form: block-v1:{course_id}+type@vertical+block@{vertical_id}
+        parent = course_blocks.get_parents(record.module_state_key)[0]
+        vertical_id = str(parent).split("@")[-1]
+
+        vertical_data.add(vertical_id)
+    if len(vertical_data) < 1:
+        return []
+
+    vertical_to_show = random.sample(vertical_data, 1)[0]
+    review_course_id = REVIEW_COURSE_MAPPING[str(current_course)]
+    return (TEMPLATE_URL.format(course_id=review_course_id,
+                    type='vertical', xblock_id=vertical_to_show))
+
+def enroll_user(user, current_course):
+    '''
+    If the user is not enrolled in the review version of the course,
+    they are unable to see any of the problems. This ensures they
+    are enrolled so they can see review problems.
+
+    Parameters:
+        user (User): the current user interacting with the review xBlock
+        current_course (CourseLocator): The course the learner is currently in
+    '''
+    enrollment_course_id = ENROLLMENT_COURSE_MAPPING[str(current_course)]
+    enrollment_status = get_enrollment(user.username, enrollment_course_id)
+    if not enrollment_status:
+        add_enrollment(user.username, enrollment_course_id)
+    elif not enrollment_status['is_active']:
+        update_enrollment(user.username, enrollment_course_id, is_active=True)
+
+def delete_state_of_review_problem(user, current_course, problem_id):
+    '''
+    Deletes the state of a review problem so it can be used infinitely
+    many times.
+
+    Parameters:
+        user (User): the current user interacting with the review xBlock
+        current_course (CourseLocator): The course the learner is currently in
+        problem_id (str?): The problem id whose state should be cleared
+    '''
+    pass
+
+def get_correctness_and_attempts(state):
+    '''
+    From the state of a problem from the Coursware Student Module,
+    determine if the learner correctly answered it initially and
+    the number of attempts they had for the original problem
+    
+    Parameter:
+        state (dict): The state of a problem
+
+    Returns a tuple of (correctness, attempts)
+        correctness (str): 'correct' or 'incorrect'
+        attempts (int): 0 if never attempted, else number of times attempted
+    Catches KeyError if learner never attempted the problem
```

Comment:
s/Catches/Raises  - but perhaps you want to implement your own exception for this case? Meaning you'd catch KeyError here and raise `ProblemNotAttemptedError` or the like instead.

Reply:
This function actually doesn't catch or raise a KeyError anymore. This comment was from before when I would catch it, but I changed the code to have an if statement for if 'attempts' in state. Let me know if you have thoughts on if I should raise some error of my own or if my if statement is fine.